Flashcards.DouglasFir/Flashcards/Application/AppStudySession.cs

```diff
@@ -0,0 +1,209 @@
+        if (selectedStack == null)
+        {
+            _manageStacksHelper.HandleNoStacksFound();
+            return;
+        }
+
+        IEnumerable<FlashCardDto> flashCards = _flashCardDao.GetAllFlashCardsByStackId(selectedStack);
+        if (flashCards == null)
+        {
+            _manageStacksHelper.HandleNoFlashCardsFound();
+            return;
+        }
+
+        StartStudySession(selectedStack, flashCards);
+    }
+
+    private void StartStudySession(StackDto stack, IEnumerable<FlashCardDto> flashCards)
+    {
+        int count = 0;
+        int score = 0;
+
+        do
+        {
+            FlashCardDto flashCard = flashCards.ElementAt(count);
+            AnsiConsole.Clear();
+            Utilities.DisplayPageHeader("Study Session");
+            Utilities.DisplayFlashCardFront(flashCard);
+
+            var response = _inputHandler.PromptForNonEmptyString("Enter your response:");
+            
+            if (Utilities.StringTrimLower(response) == Utilities.StringTrimLower(flashCard.Back!))
+            {
+                Utilities.DisplaySuccessMessage("Correct!");
+                score++;
+            }
+            else
+            {
+                Utilities.DisplayWarningMessage("Incorrect!");
+            }
+
+            Utilities.DisplayInformationConsoleMessage($"Correct Answer: {flashCard.Back}");
+            Utilities.DisplaySuccessMessage($"Score: {score}/{count + 1}");
+            _inputHandler.PauseForContinueInput();
+            count++;
+
+        } while (count < flashCards.Count());
+
+        HandleStudySessionEnd(stack, score);
+    }
+
+    private void HandleStudySessionEnd(StackDto stack, int score)
+    {
+        AnsiConsole.Clear();
+        Utilities.DisplaySuccessMessage("Study session complete!");
+        Utilities.DisplayInformationConsoleMessage($"Final Score: {score}");
+        StudySessionDto studySession = new StudySessionDto(stack.StackID, score);
+
+        try
+        {
+            _studySessionDao.InsertNewStudySession(studySession);
+        }
+        catch (Exception ex)
+        {
+            Utilities.DisplayExceptionErrorMessage("Error saving study session.", ex.Message);
+        }
+
+        _inputHandler.PauseForContinueInput();
+    }
+
+    private void HandleViewPreviousStudySessionsSelection()
+    {
+        AnsiConsole.Clear();
+        int year = _inputHandler.PromptForPositiveInteger("Please enter a year to view study sessions for:");
+        var data = _studySessionDao.GetStudySessionReportData(year);
+        if (data == null || !data.Any())
+        {
+            AnsiConsole.WriteLine("No data available for the selected year.");
+            _inputHandler.PauseForContinueInput();
+            return;
+        }
+
+        DisplayStudySessionReport(data);
+    }
+
+    private void DisplayStudySessionReport(IEnumerable<dynamic> data)
```

Comment:
I would suggest to only use `dynamic` when really needed. Here you know what the data will look like.